sdks/python/apache_beam/runners/interactive/interactive_environment.py

```diff
@@ -357,19 +363,47 @@ def get_cache_manager(self, pipeline, create_if_absent=False):
     given pipeline. If the pipeline is absent from the environment while
     create_if_absent is True, creates and returns a new file based cache
     manager for the pipeline."""
+    if self._is_in_ipython:
+      warnings.filterwarnings(
+          'ignore',
+          'options is deprecated since First stable release. References to '
+          '<pipeline>.options will not be supported',
+          category=DeprecationWarning)
+
     cache_manager = self._cache_managers.get(str(id(pipeline)), None)
+    if isinstance(pipeline, Pipeline):
+      if hasattr(pipeline.runner, '_underlying_runner'):
+        pipeline_runner = pipeline.runner._underlying_runner
+      else:
+        pipeline_runner = pipeline.runner
+    else:
+      pipeline_runner = None
     if not cache_manager and create_if_absent:
-      from apache_beam.runners.interactive import interactive_beam as ib
-      if ib.options.cache_root:
-        #TODO(victorhc): Handle the case when the path starts with "gs://"
-        if ib.options.cache_root.startswith("gs://"):
-          raise ValueError("GCS paths are not currently supported.")
-        cache_dir = tempfile.mkdtemp(dir=ib.options.cache_root)
+      cache_root = self.options.cache_root
+      if cache_root:
+        if cache_root.startswith('gs://'):
+          cache_dir = self._get_gcs_cache_dir(pipeline, cache_root)
+        else:
+          cache_dir = tempfile.mkdtemp(dir=cache_root)
+          if not isinstance(pipeline_runner, direct_runner.DirectRunner):
+            _LOGGER.warning(
+                'A local cache directory has been specified while '
+                'not using DirectRunner. It is recommended to cache into a '
+                'GCS bucket instead.')
       else:
-        cache_dir = tempfile.mkdtemp(
-            suffix=str(id(pipeline)),
-            prefix='it-',
-            dir=os.environ.get('TEST_TMPDIR', None))
+        staging_location = pipeline.options.get_all_options(
+        )['staging_location']
+        if isinstance(pipeline_runner, DataflowRunner) and staging_location:
+          cache_dir = self._get_gcs_cache_dir(pipeline, staging_location)
+          _LOGGER.warning(
```

Comment:
Maybe make this info level.

Reply:
I think that would make sense. I've changed this now.